code/centauri/light-clients/ics10-grandpa/src/client_def.rs

```diff
@@ -94,24 +97,76 @@ where
 				.map_err(Error::GrandpaPrimitives)?;
 			},
 			ClientMessage::Misbehaviour(misbehavior) => {
-				// first off is the number of equivocations >= 1/3?
-				if misbehavior.equivocations.len() < (client_state.current_authorities.len() / 3) {
+				let first_proof = misbehavior.first_finality_proof;
+				let second_proof = misbehavior.second_finality_proof;
+				if first_proof.block == second_proof.block {
+					Err(Error::Custom("Both finality proofs are for the same block".to_string()))?
+				}
+
+				let first_header = first_proof.unknown_headers.first().ok_or_else(|| {
+					Error::Custom("No headers in first finality proof".to_string())
+				})?;
+
+				let second_header = second_proof.unknown_headers.first().ok_or_else(|| {
+					Error::Custom("No headers in second finality proof".to_string())
+				})?;
+
+				if first_header.number != second_header.number {
+					Err(Error::Custom(
+						"First and second finality proofs are for different blocks".to_string(),
+					))?
+				}
```

Comment:
nahhh, they can be for diffferent blocks

Reply:
1. Check that the header with the lowest height (`unknown_headers.iter().min_by_key(|h| h.number())`) for both finality proofs exist on-chain.
2. assert that they both describe two distinct chains
3. verify that both justifications are valid

Reply:
Yeah, forgot to remove